src/Expr.js

```diff
@@ -169,35 +225,115 @@ var exprToString = function(expr, caller) {
 
   if ('object' in expr) return printObject(expr['object'])
 
-  // Versioned queries/lambdas will have an api_version field.
-  // We want to prevent it from being parsed and displayed as:
-  // Query(ApiVersion("3", "X", Var("X")))
-  var keys = Object.keys(expr).filter(
-    expression => expression !== 'api_version'
-  )
-  var fn = keys[0]
+  if ('select' in expr) {
+    return (
+      'Select(' +
+      exprToString(expr['select']) +
+      ', ' +
+      exprToString(expr['from']) +
+      ')'
+    )
+  }
 
-  // For FQL functions with special formatting concerns, we
-  // use the specialCases object above to define their casing.
-  if (fn in specialCases) fn = specialCases[fn]
+  if ('containsstr' in expr) {
+    return (
+      'ContainsStr(' +
+      exprToString(expr['containsstr']) +
+      ', ' +
+      exprToString(expr['search']) +
+      ')'
+    )
+  }
 
-  fn = fn
-    .split('_')
-    .map(function(str) {
-      return str.charAt(0).toUpperCase() + str.slice(1)
-    })
-    .join('')
+  if ('filter' in expr) {
+    return (
+      'Filter(' +
+      exprToString(expr['collection']) +
+      ', ' +
+      exprToString(expr['filter']) +
+      ')'
+    )
+  }
+
+  if ('lambda' in expr) {
+    return (
+      'Lambda(' +
+      exprToString(expr['lambda']) +
+      ', ' +
+      exprToString(expr['expr']) +
+      ')'
+    )
+  }
+
+  if ('foreach' in expr) {
+    return (
+      'Foreach(' +
+      exprToString(expr['collection']) +
+      ', ' +
+      exprToString(expr['foreach']) +
+      ')'
+    )
+  }
+
+  if ('if' in expr) {
+    return (
+      'If(' +
+      exprToString(expr['if']) +
+      ', ' +
+      exprToString(expr['then']) +
+      ', ' +
+      exprToString(expr['else']) +
+      ')'
+    )
+  }
+
+  if ('call' in expr) {
+    return (
+      'Call(' +
+      exprToString(expr['call']) +
+      ', ' +
+      exprToString(expr['arguments']) +
+      ')'
+    )
+  }
+
+  if ('databases' in expr) {
+    if (!expr['databases']) return 'Databases()'
+
+    return 'Databases(' + exprToString(expr['databases']) + ')'
+  }
+
+  if ('collections' in expr) {
+    if (!expr['collections']) return 'Collections()'
+
+    return 'Collections(' + exprToString(expr['collections']) + ')'
+  }
+
+  if ('documents' in expr) {
+    if (!expr['documents']) return 'Documents()'
+    return 'Documents(' + exprToString(expr['documents']) + ')'
+  }
+
+  if ('map' in expr) {
+    return (
+      'Map(' +
+      exprToString(expr['collection']) +
+      ', ' +
+      exprToString(expr['map']) +
+      ')'
+    )
+  }
+
+  var keys = Object.keys(expr)
+  var fn = keys[0]
+  fn = convertToCamelCase(fn)
 
   var args = keys.map(function(k) {
     var v = expr[k]
     return exprToString(v, fn)
   })
 
-  var shouldReverseArgs = ['filter', 'map', 'foreach'].some(function(fn) {
-    return fn in expr
-  })
-
-  if (shouldReverseArgs) {
+  if (shouldReverseArgs(expr)) {
```

Comment:
now that you handled all the edge case, that function call and the function itself is not necessary